lib/utils/validation.ts

```diff
@@ -75,9 +75,13 @@ export function validateUuid(input: string): boolean {
 	return validateLongUuid(input) || validateShortUuid(input);
 }
 
-export function parseAsInteger(input: string, paramName?: string) {
+export function looksLikeInteger(input: string) {
 	// Allow only digits, no leading 0
-	if (!/^(0|[1-9][0-9]*)$/.test(input)) {
+	return !/^(0|[1-9][0-9]*)$/.test(input);
```

Comment:
```suggestion
	return !/^(?:0|[1-9][0-9]*)$/.test(input);
```

Reply:
`?:` introduces a [non-capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges) as opposed to a capturing group. Out of interest, what would be the advantage of using it here, with `.test()`?

Reply:
There's a performance cost to a capturing group (you have to track the match, store it in an array, etc), it's possible in a `.test` it makes no difference (I'd hope so) but I use non-capturing groups as a best practice wherever we don't actually care about the contents

Reply:
Updated